internal/integration/integration_test.go

```diff
@@ -493,6 +493,87 @@ func testCLISchemaApplyAutoApprove(t T, h string, dsn string, args ...string) {
 	require.NotNil(t, u)
 }
 
+func testCLISchemaApplyFromMigrationDir(t T) {
+	const (
+		DB    = "apply_migration_dir"
+		DevDB = DB + "_dev"
+	)
+	t.dropSchemas(DB, DevDB)
+	t.migrate(&schema.AddSchema{S: schema.New(DB)}, &schema.AddSchema{S: schema.New(DevDB)})
+	defer t.migrate(&schema.DropSchema{S: schema.New(DB)}, &schema.DropSchema{S: schema.New(DevDB)})
+
+	users, err := t.driver().SchemaDiff(schema.New(""), schema.New("").AddTables(t.users()))
+	require.NoError(t, err)
+	usersT := t.users()
+	usersT.Name = "users_2"
+	users2, err := t.driver().SchemaDiff(schema.New(""), schema.New("").AddTables(usersT))
+	require.NoError(t, err)
+
+	addUsers, err := t.driver().PlanChanges(context.Background(), "", users)
+	require.NoError(t, err)
+	addUsers2, err := t.driver().PlanChanges(context.Background(), "", users2)
+	require.NoError(t, err)
+
+	var (
+		fn = func(c []*migrate.Change) string {
+			var buf strings.Builder
+			for _, c := range c {
+				buf.WriteString(c.Cmd)
+				buf.WriteString("\n")
+			}
+			return buf.String()
+		}
+		one = fn(addUsers.Changes)
+		two = fn(addUsers2.Changes)
+	)
+
+	p := t.TempDir()
+	require.NoError(t, os.WriteFile(filepath.Join(p, "1.sql"), []byte(one), 0644))
+	require.NoError(t, os.WriteFile(filepath.Join(p, "2.sql"), []byte(two), 0644))
+
+	require.NoError(t, initCLI())
+	require.NoError(t, exec.Command(
+		"go", "run", "ariga.io/atlas/cmd/atlas",
+		"migrate", "hash",
+		"--dir", "file://"+p,
+	).Run())
+
+	// All versions - must contain all migration files.
+	cmd := exec.Command(
+		"go", "run", "ariga.io/atlas/cmd/atlas",
+		"schema", "apply",
+		"-u", t.url(DB),
+		"--to", "file://"+p,
+		"--dev-url", t.url(DevDB),
+		"--dry-run",
+	)
+	stdout, stderr := bytes.NewBuffer(nil), bytes.NewBuffer(nil)
+	cmd.Stderr = stderr
+	cmd.Stdout = stdout
+	require.NoError(t, cmd.Run(), stderr.String(), stdout.String())
+	require.Empty(t, stderr.String())
+	require.Contains(t, stdout.String(), one)
+	require.Contains(t, stdout.String(), two)
```

Comment:
You can use CombinedOutput: https://github.com/ariga/atlas/blob/b1a3d4b5868a4e59f306548df63b86ffcec1c449/internal/integration/integration_test.go#L186-L193